File: benchmark-java/src/main/java/com/benchmark/config/CouchbaseConfig.java
package com.benchmark.config;

import com.couchbase.client.java.Cluster;
import com.couchbase.client.java.ClusterOptions;
import com.couchbase.client.java.Collection;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.time.Duration;

@Configuration
public class CouchbaseConfig {

    @Value("${spring.couchbase.connection-string}")
    private String connectionString;

    @Value("${spring.couchbase.username}")
    private String username;

    @Value("${spring.couchbase.password}")
    private String password;

    @Value("${spring.couchbase.bucket-name}")
    private String bucketName;

    @Bean
    public Cluster couchbaseCluster() {
        return Cluster.connect(
            connectionString,
            ClusterOptions.clusterOptions(username, password).environment(env -> {
                env.timeoutConfig().kvScanTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().kvTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().kvDurableTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().queryTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().viewTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().analyticsTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().connectTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().disconnectTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().managementTimeout(Duration.ofSeconds(100));
                env.timeoutConfig().searchTimeout(Duration.ofSeconds(100));

            })
        );
    }

    @Bean
    public Collection couchbaseCollection(Cluster cluster) {
        return cluster.bucket(bucketName).defaultCollection();
    }
} 

File: benchmark/config/couchbase_config.py
from typing import Dict, Any
from couchbase.cluster import Cluster, ClusterOptions
from couchbase.auth import PasswordAuthenticator

class CouchbaseConfig:
    def __init__(self, config: Dict[str, Any]):
        self.host = config.get('COUCHBASE_HOST', 'localhost')
        self.port = int(config.get('COUCHBASE_PORT', 8091))
        self.bucket = config.get('COUCHBASE_BUCKET', 'benchmark')
        self.username = config.get('COUCHBASE_USERNAME', 'Administrator')
        self.password = config.get('COUCHBASE_PASSWORD', 'password')
        self.cluster = None
        self.bucket_obj = None
        self.collection = None

    def connect(self) -> None:
        """Initialize Couchbase connection"""
        connection_string = f'couchbase://{self.host}:{self.port}'
        auth = PasswordAuthenticator(self.username, self.password)
        options = ClusterOptions(auth)
        
        self.cluster = Cluster(connection_string, options)
        self.bucket_obj = self.cluster.bucket(self.bucket)
        self.collection = self.bucket_obj.default_collection()

    def disconnect(self) -> None:
        """Close Couchbase connection"""
        if self.cluster:
            self.cluster.disconnect()
            self.cluster = None
            self.bucket_obj = None
            self.collection = None 